### PR TITLE
fix(eslint): allow nested ternaries

### PR DESCRIPTION
**Pull request by stevensacks**
JSX uses nested ternaries.

Another thing to consider is if we use Prettier, Prettier makes nested ternaries a lot nicer looking.

### CODE REVIEW

**Comment by PixnBits** [comment]
>Prettier makes nested ternaries a lot nicer looking.

Heh, this was the straw that pushed me over the threshold of removing prettier from my projects 😅

File: index.js [hunk 1]
```diff
@@ -105,6 +105,12 @@ module.exports = {
     // regular expressions.
     'unicorn/no-unsafe-regex': 'error',
 
+    // Nested ternaries are commonly used in JSX
+    // https://eslint.org/docs/rules/no-nested-ternary
+    'no-nested-ternary': 'off',
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-nested-ternary.md
+    'unicorn/no-nested-ternary': 'off',
```
Comment (PixnBits) [inline index.js:112]: 'unicorn/no-nested-ternary' is a bit different from 'no-nested-ternary'
>~Improved~ [Different] version of the no-nested-ternary ESLint rule, which allows cases where the nested ternary is only one level and wrapped in parens.

Might be good to explain why 'unicorn/no-nested-ternary' is disabled too? Presumably due to nesting level limitation? If that's true, also include why no limit on depth _might_ be okay. Be sure to address the "Nesting ternary expressions can make code more difficult to understand" claim of 'no-nested-ternary' and how this is mitigated.

**Comment by guym4c** [comment]
IMO, a nested ternary is a sign that you need to split out a branch to improve readability and remove the nesting

**Comment by PixnBits** [comment]
@guym4c sometimes, but other times I'd argue against that. Example from some webpack configuration I needed:
```javascript
let someThing;
if (condition === 'first possible value') {
  someThing = 'a';
} else if (condition === 'second possible value') {
  someThing = 'b';
} else {
  someThing = 'c';
}
```
vs
```javascript
const someThing = 
  condition === 'first possible value' ? 'a' :
  condition === 'second possible value' ? 'b' :
  'c';
```
I'm communicated that I don't want this value to change later on in the file (and guard against it)
This _could_ also be accomplished via an immediately invoked function
```javascript
const someThing = (() => {
  if (condition === 'first possible value') {
    return 'a';
  }
  if (condition === 'second possible value') {
    return 'b';
  }
  return 'c';
})();
// or, if you're fine w/o explicit if/else bodies:
const someThing = (() => {
  if (condition === 'first possible value') return 'a';
  if (condition === 'second possible value') return 'b';
  return 'c';
})();
// or
const someThing = (() => {
  switch (condition) {
    case: 'first possible value':
      return 'a';
    case: 'second possible value':
      return 'b';
    default:
      return 'c';
})();
```
but that readability seems less helpful and isn't nearly as easy for static code parsers to analyze

File: index.js (hunk 1 above)
Comment (stevensacks) [reply index.js:123]: It's a redundant rule.

**Comment by stevensacks** [comment]
If you don't want to use ternaries, don't use them. Nobody is forcing you to. But for the people who do like using ternaries, let Prettier do its job and make them, well, Prettier.

**Comment by PixnBits** [comment]
Nested ternaries should probably be an exception instead of the rule. @10xLaCroixDrinker I move to decline this PR

**Pull request closed without merging**